components/api/foos/foo_query.go

```diff
@@ -0,0 +1,25 @@
+package foos
+
+import (
+	"github.com/jmoiron/sqlx"
+
+	"github.com/graphql-go/graphql"
+)
+
+//NewFooQueryField makes a new GraphQL field for querying the foo type.
+func NewFooQueryField(db *sqlx.DB) *graphql.Field {
+	fooType := makeFooType()
+	return &graphql.Field{
+		Name: "foos",
+		Type: graphql.NewList(fooType),
+		Args: graphql.FieldConfigArgument{
+			"id": &graphql.ArgumentConfig{
+				Type: graphql.Int,
+			},
+		},
+		Resolve: func(params graphql.ResolveParams) (interface{}, error) {
+			resolver := newLoadResolver(db)
```

Comment:
I want to experiment with replacing this resolver struct with functions. The structs aren't adding that much value and I think there's some benefits gained (parallel queries) if a function is returned here.